src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -402,19 +413,20 @@ private void Cancel()
             }
 
             // Returns whether the waiter should be signalled or not.
-            private (bool signalWaiter, bool sendReset) CancelResponseBody()
+            private (bool signalWaiter, bool sendReset, bool complete) CancelResponseBody()
             {
                 Debug.Assert(Monitor.IsEntered(SyncObject));
 
                 bool sendReset = false;
+                bool complete = false;
 
                 if (_responseCompletionState == StreamCompletionState.InProgress)
                 {
                     _responseCompletionState = StreamCompletionState.Failed;
                     if (_requestCompletionState != StreamCompletionState.InProgress)
                     {
                         sendReset = true;
-                        Complete();
+                        complete = true;
```

Comment:
Does it even differ from `sendReset`? Do we need an extra variable?

Reply:
Yes, you are right. I removed the redundant variable.